simplified-app-shared/src/main/java/org/nypl/simplified/app/reader/ReaderActivity.java

```diff
@@ -490,15 +506,13 @@ else if (id == 1) {
   {
     ReaderActivity.LOG.debug("received book location: {}", l);
 
-    //TODO see if I really need this.current_location as an ivar
     this.current_location = l;
 
     final SimplifiedReaderAppServicesType rs = Simplified.getReaderAppServices();
-    final ReaderBookmarksType bm = rs.getBookmarks();
-    //TODO could book_id actually be null?
+    final ReaderBookmarksSharedPrefsType bm = rs.getBookmarks();
     final BookID in_book_id = NullCheck.notNull(this.book_id);
 
-//    bm.setBookmark(in_book_id, this.current_location);
+    bm.saveReadingPosition(in_book_id, this.current_location);
```

Comment:
The app was using "bookmarks" to mean "last place you were reading" so I've updated that to add more separation and clarity between the two.